net/isc-dhcp/files/dhcpd.init

```diff
@@ -302,13 +309,33 @@ start_service() {
 	procd_close_instance
 }
 
-boot() {
-	DHCPD_BOOT=1
-	start "$@"
+reload_service() {
+	rc_procd start_service "$@"
+	prodcd_send_signal dhcpd "$@"
 }
 
-service_triggers()
-{
-	procd_add_reload_trigger "dhcp"
-	procd_add_raw_trigger "interface.*" 3000 /etc/init.d/dhcpd reload
+add_interface_trigger() {
+	local cfg=$1
+	local trigger ignore
+
+	config_get trigger "$cfg" interface
+	config_get_bool ignore "$cfg" ignore 0
+
+	if [ -n "$trigger" -a $ignore -eq 0 ] ; then
+		procd_add_reload_interface_trigger "$trigger"
+	fi
 }
+
+service_triggers() {
+	if [ -n "$DHCPD_BOOT" ] ; then
+		# Make the first start robust to slow interfaces; wait a while
+		procd_add_raw_trigger "interface.*.up" 5000 /etc/init.d/dhcpd restart
```

Comment:
I have struggled with these trigger lines on other packages. DHCPv6 ISP can generate a lot of IFUP like events when broadcasting minor link value updates. You seem to boot-start DHCP processing after the first interface actually is online, then only re-trigger on DHCP enabled interfaces exclusively. This should work but I wonder how to certainly test that a startup is not stalled and that the daemon isn't purged/reloaded to frequently.

Reply:
We could use Qemu and OpenvSwitch to come up with a test harness...